src/lib/operators/index_scan.cpp

```diff
@@ -187,18 +191,19 @@ RowIDPosList IndexScan::_scan_chunk(const ChunkID chunk_id) {
   }
 
   DebugAssert(_in_table->type() == TableType::Data, "Cannot guarantee single chunk PosList for non-data tables.");
-  matches_out.guarantee_single_chunk();
 
-  const auto current_matches_size = matches_out.size();
-  const auto final_matches_size = current_matches_size + static_cast<size_t>(std::distance(range_begin, range_end));
-  matches_out.resize(final_matches_size);
+  auto& chunk_offsets = singleChunkPosList->get_offsets();
+  const auto previous_matches_size = chunk_offsets.size();
+  const auto matches_size = previous_matches_size + static_cast<size_t>(std::distance(range_begin, range_end));
+  chunk_offsets.resize(matches_size);
 
-  for (auto matches_position = current_matches_size; matches_position < final_matches_size; ++matches_position) {
-    matches_out[matches_position] = RowID{chunk_id, *range_begin};
+  for (auto matches_position = previous_matches_size; matches_position < matches_size; ++matches_position) {
+    chunk_offsets[matches_position] = *range_begin;
     range_begin++;
   }
 
-  return matches_out;
+  // TODO: Maybe it is more easy to make the singleChunkPosList here, and previously only edit a chunk_offset vector
```

Comment:
+1, that's how I would have expected it to be designed anyway. Is there a reason not to do this? @XPerianer